CalibTracker/SiStripLorentzAngle/plugins/SiStripLorentzAnglePCLMonitor.cc

```diff
@@ -0,0 +1,499 @@
+      const auto locB = magField.inTesla(stripDet->surface().position());
+      c_localB.push_back(locB.mag());
+      c_BdotY.push_back(stripDet->surface().toLocal(locB).y());
+      const auto drift = shallow::drift(stripDet, magField, lorentzAngle);
+      c_driftx.push_back(drift.x());
+      c_drifty.push_back(drift.y());
+      c_driftz.push_back(drift.z());
+      c_lorentzAngle.push_back(lorentzAngle.getLorentzAngle(detid));
+      iHists_.la_db_[detid] = lorentzAngle.getLorentzAngle(detid);
+      iHists_.moduleLocationType_[detid] = this->moduleLocationType(detid, tTopo);
+    }
+  }
+
+  // Sorted DetId list gives max performance, anything else is worse
+  std::sort(c_rawid.begin(), c_rawid.end());
+
+  // initialized the hash map
+  m_hash = SiStripHashedDetId(c_rawid);
+
+  //reserve the size of the vector
+  iHists_.h2_ct_w_m_.reserve(c_rawid.size());
+  iHists_.h2_ct_var2_m_.reserve(c_rawid.size());
+  iHists_.h2_ct_var3_m_.reserve(c_rawid.size());
+
+  iHists_.h2_t_w_m_.reserve(c_rawid.size());
+  iHists_.h2_t_var2_m_.reserve(c_rawid.size());
+  iHists_.h2_t_var3_m_.reserve(c_rawid.size());
+}
+
+std::string SiStripLorentzAnglePCLMonitor::moduleLocationType(const uint32_t& mod, const TrackerTopology* tTopo) {
+  const SiStripDetId detid(mod);
+  std::string subdet = "";
+  unsigned int layer = 0;
+  if (detid.subDetector() == SiStripDetId::TIB) {
+    subdet = "TIB";
+    layer = tTopo->layer(mod);
+  }
+
+  if (detid.subDetector() == SiStripDetId::TOB) {
+    subdet = "TOB";
+    layer = tTopo->layer(mod);
+  }
+
+  std::string type = (detid.stereo() ? "s" : "a");
+  std::string d_l_t = Form("%s_L%d%s", subdet.c_str(), layer, type.c_str());
+
+  if (layer == 0)
+    return subdet;
+  return d_l_t;
+}
+
+// ------------ method called for each event  ------------
+void SiStripLorentzAnglePCLMonitor::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace edm;
+
+  // return immediately if the field is not consistent!
+  if (mismatchedBField_)
+    return;
+
+  if (mismatchedLatency_)
+    return;
+
+  edm::Handle<edm::View<reco::Track>> tracks;
+  iEvent.getByToken(m_tracks_token, tracks);
+  edm::Handle<TrajTrackAssociationCollection> trajTrackAssociations;
+  iEvent.getByToken(m_association_token, trajTrackAssociations);
+
+  LogDebug(moduleDescription().moduleName()) << "I AM IN EVENT" << iEvent.id() << std::endl;
+
+  std::vector<OnTrackCluster> clusters{};
+
+  // first collect all the clusters
+  for (const auto& assoc : *trajTrackAssociations) {
+    const auto traj = assoc.key.get();
+    const auto track = assoc.val.get();
+
+    iHists_.h1_["track_pt"]->Fill(track->pt());
+    iHists_.h1_["track_eta"]->Fill(track->eta());
+    iHists_.h1_["track_phi"]->Fill(track->phi());
+    iHists_.h1_["track_validhits"]->Fill(track->numberOfValidHits());
+    iHists_.h1_["track_chi2ndof"]->Fill((track->chi2() / track->ndof()));
```

Comment:
You could define some
```
    const auto normChi2 = track->chi2() / track->ndof();
```
or even 
```
    const auto normChi2 = track->ndof() > 0 ? track->chi2() / track->ndof() : -1.;
```
and use consistently to fill the histos

Reply:
Addressed at https://github.com/mmusich/cmssw/commit/19bf615f8649660683e0652f5289fb7f3e5d0d93